zebrad/src/commands/start.rs

```diff
@@ -60,6 +61,12 @@ impl StartCmd {
     }
 }
 
+async fn on_ctrl_c() {
+    tokio::signal::ctrl_c().await.unwrap();
+    let _ = crate::application::FLAME_GUARD.lock().unwrap().take();
```

Comment:
Issue: Destructors on the current stack don't run after exit:
https://doc.rust-lang.org/std/process/fn.exit.html

Suggestion: call a function that explicitly writes the flame graph file. The current code obscures graph creation inside an implicit destructor, on an exiting stack, inside a type-erased, multi-layer object. Without tests, this kind of hidden functionality is likely to be broken in a refactor.

Suggestion: add an acceptance test that makes sure that the destructor actually runs:
https://github.com/ZcashFoundation/zebra/blob/main/zebrad/tests/acceptance.rs#L175

```suggestion
    // Make sure the destructor runs before exit
    {
        let _ = crate::application::FLAME_GUARD.lock().unwrap().take();
    }
```

Reply:
Also, if the flame graphs aren't written on panic, we should document that fact.

Reply:
they should be written in a panic so long as the panic is non aborting. I can also probably add features to color-eyre so we can add custom callbacks to the panic handler so we can do things like call important drop impls on panic even when we have panic = abort.

Reply:
Also @teor2345 the scoping of the `take` is redundant in rust. Binding a variable to `_` is well defined behaviour and is the same as _not_ making a binding, rather than making a binding without a name. So binding a variable to `_` always immediately drops the variable.